Modules/posixmodule.c

```diff
@@ -4139,8 +4141,8 @@ _listdir_windows_no_opendir(path_t *path, PyObject *list)
         int error = GetLastError();
         if (error == ERROR_FILE_NOT_FOUND)
             goto exit;
-        Py_DECREF(list);
-        list = path_error(path);
+        path_error(path);
+        Py_SETREF(list, NULL);
```

Comment:
Py_CLEAR() can be used. Same remark below.

Oh wow, ``list = path_error(path);`` code is wild! But it works :-D